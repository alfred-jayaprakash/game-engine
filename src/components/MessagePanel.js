import React from 'react';
import {ListGroup, ListGroupItem} from 'reactstrap';

const MessagePanel = props => {
  return (
    <div className="centered-form__box">
      Some empty area to display your words
      <ListGroup>
        {props.data.map (answer => (
          <ListGroupItem key="{answer}">{answer}</ListGroupItem>
        ))}
      </ListGroup>
    </div>
  );
};

export default MessagePanel;
