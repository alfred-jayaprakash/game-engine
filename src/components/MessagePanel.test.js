import React from 'react';
import {render, fireEvent, screen, act, waitFor} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {shallow} from 'enzyme';
import MessagePanel from './MessagePanel';

let container;
let data;

describe ('Snapshot tests', () => {
  beforeAll (() => {
    //ARRANGE
    data = ['one', 'two', 'three'];
  });

  it ('should render correctly without issues', () => {
    const component = shallow (<MessagePanel data={data} />);

    expect (component).toMatchSnapshot ();
  });
});

describe ('Functional tests', () => {
  beforeAll (() => {
    //ARRANGE
  });

  beforeEach (() => {
    container = document.createElement ('div');
    document.body.appendChild (container);
  });

  afterEach (() => {
    document.body.removeChild (container);
    container = null;
  });

  it ('initial screen should load correctly', () => {
    data = ['one', 'two', 'three'];
    render (<MessagePanel data={data} />, container);
  });
});
