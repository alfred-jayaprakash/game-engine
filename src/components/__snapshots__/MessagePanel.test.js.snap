// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests should render correctly without issues 1`] = `
<Row
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <ListGroup
      horizontal={false}
      tag="ul"
    >
      <ListGroupItem
        key="{answer}"
        tag="li"
      >
        one
      </ListGroupItem>
      <ListGroupItem
        key="{answer}"
        tag="li"
      >
        two
      </ListGroupItem>
      <ListGroupItem
        key="{answer}"
        tag="li"
      >
        three
      </ListGroupItem>
    </ListGroup>
  </Col>
</Row>
`;
